plutus-core/plutus-ir/test/datatypes/listMatch.golden

```diff
@@ -1,46 +1,23 @@
 (program 1.0.0
   [
     [
-      [
-        (lam
-          Nil_i0
-          (all a_i0 (type) [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) a_i1])
-          (lam
-            Cons_i0
-            (all a_i0 (type) (fun a_i1 (fun [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) a_i1] [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) a_i1])))
-            (lam
-              match_List_i0
-              (all a_i0 (type) (fun [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) a_i1] (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) a_i2] out_List_i1)) out_List_i1)))))
-              [
-                [
-                  {
-                    [
-                      { match_List_i1 (all a_i0 (type) (fun a_i1 a_i1)) }
-                      { Nil_i3 (all a_i0 (type) (fun a_i1 a_i1)) }
-                    ]
-                    (all a_i0 (type) (fun a_i1 a_i1))
-                  }
-                  (abs a_i0 (type) (lam x_i0 a_i2 x_i1))
-                ]
-                (lam
-                  head_i0
-                  (all a_i0 (type) (fun a_i1 a_i1))
-                  (lam
-                    tail_i0
-                    [(lam a_i0 (type) (ifix (lam List_i0 (fun (type) (type)) (lam a_i0 (type) (all out_List_i0 (type) (fun out_List_i1 (fun (fun a_i2 (fun [List_i3 a_i2] out_List_i1)) out_List_i1))))) a_i1)) (all a_i0 (type) (fun a_i1 a_i1))]
-                    head_i2
-                  )
-                )
-              ]
+      {
```

Comment:
Separately: we should probably run these test cases with optimization off, since they're more about testing that e.g. the datatype compilation pass does the right thing, which is harder to see after optimization! Or possibly I should just get them to run the pass that we actually care about :thinking:

Reply:
Hmm, on the other hand I found those data type passes good to find bugs in the simplifier, because we run the simplifier on slightly bigger code samples, and IIRC they include type-checking pass after simplification (the `transform/` passes do not include type checking passes).

Reply:
@michaelpj do you want to turn of simplifier for these test cases as well?

Reply:
For these ones I think we should rather just run the specific pass that we're testing. At the moment it's running the whole compiler, which is a bit crude anyway. I can fix those up later.